### DISCUSSION TITLE
Feature Request: Support for midi in RIFF container

### DISCUSSION THREAD

**Discussion by maxmitti** [FluidSynth/fluidsynth#1094] [Ideas]
Some midi files are stored in a RIFF container.
It would be helpful if fluidsynth supported those the same way as normal midis.
Timidity apparently supports that. The fluidsynth cli says its not a midi file.

**Comment by derselbst** [comment]
Proper RMID support would require to also support DLS files, which may be bundled within the RMID. You're welcome to draft a PR for RMID support. Here is the spec:
http://web.archive.org/web/20110610135604/http://www.midi.org/about-midi/rp29spec(rmid).pdf

However, it's unlikely that incomplete RMID support which only cares about the MIDI file would be accepted.

**Reply by maxmitti** [reply]
What about RMID files that don’t actually contain a DLS? They seem to be not that uncommon.
FluidSynth could potentially support those files with little work (I guess) and error out on RMIDs with DLS.

**Reply by derselbst** [reply]
RMID is a standardized format specified by the MIDI Association. And I am not in favor of cherry-picking arbitrary points from a spec, ending up in an incomplete implementation, tempting people raise bug reports, claiming "this doesn't work, but according to the spec it should".
